pkg/mdformatter/mdformatter_bench_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package mdformatter
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"testing"
+
+	"github.com/efficientgo/tools/core/pkg/testutil"
+)
+
+func benchMdformatter(filename string, b *testing.B) {
+	file, err := os.OpenFile(filename, os.O_RDONLY, 0)
+	testutil.Ok(b, err)
+	defer file.Close()
+
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		f := New(context.Background())
+
+		b.Run(filename, func(b *testing.B) {
+			buf := bytes.Buffer{}
```

Comment:
Since you don't use this buf, there is a strong risk that compiler will optimize away the WHOLE thing (because result is unused).

 I think it's unlikely in your case, because I don't think compiler can tell if Format has side effects or not, but it's advise to create a global `testBuf` variable and use that (allocate new buffer all the time)